modin/pandas/groupby.py

```diff
@@ -128,7 +129,46 @@ def sem(self, ddof=1):
         return self._default_to_pandas(lambda df: df.sem(ddof=ddof))
 
     def mean(self, *args, **kwargs):
-        return self._apply_agg_function(lambda df: df.mean(*args, **kwargs))
+        fallback = False
+        converted_columns = {}
+        numeric_only = kwargs.get("numeric_only", False)
+        if not numeric_only:
+            for col, dt in self._query_compiler.dtypes.items():
+                if is_datetime64_any_dtype(dt):
+                    if self._df[col].hasnans:
+                        fallback = True
+                        break
+                    else:
+                        converted_columns[col] = dt
+        if fallback:
+            # We cannot use map-reduce approach because non-numeric columns are present and
+            # are requested to be processed. It happens because in map-reduce we use "sum"
+            # aggregation which always drops non-numeric columns unconditionally, no matter
+            # what arguments are specified. It is different from how "mean" handles non-numeric
+            # columns (in particular datetime types). We could use a workaround to convert
+            # datetime to int64 but it works only when NaNs aren't present in datetime columns.
+            # NaNs converted to int64 produce wrong results and have to be handled differently,
+            # so we have to resort to less efficient approach of broadcast full axis in
```

Comment:
why does `NaNs` converted to int produce wrong results? As far as I know, we can convert `NaN` to `float` only. Does backward converting of `float` -> `int` -> `datetime` causing precision loss and so incorrect results?

Reply:
Consider the following example:
```
df = pd.DataFrame({"a": [1, 1, 2, 2], "b": [np.datetime64("2001-01-01"), np.NaN, np.datetime64("2001-01-01"), np.datetime64("2002-01-01")]})
```
Here if we convert column "b" to int64 as `df.astype({"b": "int64"})` you can see that NaN in column "b" is converted into value `-9223372036854775808` which affects the final sum of values. If NaN was converted into zero, we could use this approach because effectively NaN values would be ignored. We cannot convert into float64 because pandas gives an error.

Reply:
what would it take to make this case (or just all datetimelike cases) just use the pandas implementation? https://github.com/pandas-dev/pandas/blob/main/pandas/_libs/groupby.pyx#L721